### PR TITLE
feat(chart): add helm chart for zfs local pv

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does?**:
This PR adds helm charts for zfs local pv. 
TODO: add Readme for the config flags.

**If the changes in this PR are manually verified, list down the scenarios covered:**:
```sh
$ helm install openebs --debug --dry-run -n autogen --namespace openebs deploy/helm/charts/
install.go:159: [debug] Original chart version: ""
install.go:176: [debug] CHART PATH: /home/mayadata/work/src/github.com/openebs/zfs-localpv/deploy/helm/charts

install.go:184: [debug] WARNING: This chart or one of its subcharts contains CRDs. Rendering may fail or contain inaccuracies.
NAME: openebs
LAST DEPLOYED: Mon Dec  7 11:54:24 2020
NAMESPACE: openebs
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
controller:
  annotations: {}
  componentName: openebs-zfs-controller
  nodeSelector: {}
  podAnnotations: {}
  podLabels:
    name: openebs-zfs-controller
  provisioner:
    image:
      pullPolicy: IfNotPresent
      registry: quay.io/
      repository: k8scsi/csi-provisioner
      tag: v1.6.0
    name: csi-provisioner
  resizer:
    image:
      pullPolicy: IfNotPresent
      registry: quay.io/
      repository: k8scsi/csi-resizer
      tag: v0.4.0
    name: csi-resizer
  resources: {}
  securityContext: {}
  snapshotController:
    image:
      pullPolicy: IfNotPresent
      registry: quay.io/
      repository: k8scsi/snapshot-controller
      tag: v2.0.1
    name: snapshot-controller
  snapshotter:
    image:
      pullPolicy: IfNotPresent
      registry: quay.io/
      repository: k8scsi/csi-snapshotter
      tag: v2.0.1
    name: csi-snapshotter
  tolerations: []
  updateStrategy:
    type: RollingUpdate
crd:
  enableInstall: true
imagePullSecrets: null
release:
  version: 1.1.0
role: openebs-zfs
serviceAccount:
  controller:
    create: true
    name: openebs-zfs-controller-sa
  zfsNode:
    create: true
    name: openebs-zfs-node-sa
zfsNode:
  annotations: {}
  componentName: openebs-zfs-node
  driverRegistrar:
    image:
      pullPolicy: IfNotPresent
      registry: quay.io/
      repository: k8scsi/csi-node-driver-registrar
      tag: v1.2.0
    name: csi-node-driver-registrar
  enabled: true
  labels: {}
  nodeSelector: {}
  podAnnotations: {}
  podLabels:
    name: openebs-zfs-node
  resources: {}
  securityContext: {}
  tolerations: []
  updateStrategy:
    type: RollingUpdate
zfsPlugin:
  image:
    pullPolicy: IfNotPresent
    registry: null
    repository: openebs/zfs-driver
    tag: 1.1.0
  name: openebs-zfs-plugin

HOOKS:
MANIFEST:
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ServiceAccount
apiVersion: v1
metadata:
  name: openebs-zfs-controller-sa
  namespace: openebs
---
# Source: zfs-localpv/templates/rbac.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: openebs-zfs-node-sa
  namespace: openebs
---
# Source: zfs-localpv/templates/configmap.yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: openebs-zfspv-bin
  namespace: openebs # should be the same namespace where it is getting mounted
data:
  zfs: |
    #!/bin/sh
    if [ -x /host/sbin/zfs ]; then
      chroot /host /sbin/zfs "$@"
    elif [ -x /host/usr/sbin/zfs ]; then
      chroot /host /usr/sbin/zfs "$@"
    else
      chroot /host zfs "$@"
    fi
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: openebs-zfs-provisioner-role
rules:
  - apiGroups: [""]
    resources: ["secrets"]
    verbs: ["get", "list"]
  - apiGroups: [""]
    resources: ["namespaces"]
    verbs: ["*"]
  - apiGroups: [""]
    resources: ["persistentvolumes", "services"]
    verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
  - apiGroups: [""]
    resources: ["persistentvolumeclaims"]
    verbs: ["get", "list", "watch", "update"]
  - apiGroups: [""]
    resources: ["persistentvolumeclaims/status"]
    verbs: ["update", "patch"]
  - apiGroups: ["storage.k8s.io"]
    resources: ["storageclasses", "csinodes"]
    verbs: ["get", "list", "watch"]
  - apiGroups: [""]
    resources: ["events"]
    verbs: ["list", "watch", "create", "update", "patch"]
  - apiGroups: ["coordination.k8s.io"]
    resources: ["leases"]
    verbs: ["get", "watch", "list", "delete", "update", "create"]
  - apiGroups: [""]
    resources: ["nodes"]
    verbs: ["get", "list", "watch"]
  - apiGroups: ["*"]
    resources: ["zfsvolumes", "zfssnapshots", "zfsbackups", "zfsrestores"]
    verbs: ["*"]
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: openebs-zfs-snapshotter-role
rules:
  - apiGroups: [""]
    resources: ["persistentvolumes"]
    verbs: ["get", "list", "watch"]
  - apiGroups: [""]
    resources: ["persistentvolumeclaims"]
    verbs: ["get", "list", "watch"]
  - apiGroups: ["storage.k8s.io"]
    resources: ["storageclasses"]
    verbs: ["get", "list", "watch"]
  - apiGroups: [""]
    resources: ["events"]
    verbs: ["list", "watch", "create", "update", "patch"]
  - apiGroups: [""]
    resources: ["secrets"]
    verbs: ["get", "list"]
  - apiGroups: ["snapshot.storage.k8s.io"]
    resources: ["volumesnapshotclasses"]
    verbs: ["get", "list", "watch"]
  - apiGroups: ["snapshot.storage.k8s.io"]
    resources: ["volumesnapshotcontents"]
    verbs: ["create", "get", "list", "watch", "update", "delete"]
  - apiGroups: ["snapshot.storage.k8s.io"]
    resources: ["volumesnapshots"]
    verbs: ["get", "list", "watch", "update"]
  - apiGroups: ["snapshot.storage.k8s.io"]
    resources: ["volumesnapshotcontents/status"]
    verbs: ["update"]
  - apiGroups: ["snapshot.storage.k8s.io"]
    resources: ["volumesnapshots/status"]
    verbs: ["update"]
  - apiGroups: ["apiextensions.k8s.io"]
    resources: ["customresourcedefinitions"]
    verbs: ["create", "list", "watch", "delete"]
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: openebs-zfs-driver-registrar-role
rules:
  - apiGroups: [""]
    resources: ["events"]
    verbs: ["get", "list", "watch", "create", "update", "patch"]
  - apiGroups: [""]
    resources: ["persistentvolumes", "nodes", "services"]
    verbs: ["get", "list"]
  - apiGroups: ["*"]
    resources: ["zfsvolumes", "zfssnapshots", "zfsbackups", "zfsrestores"]
    verbs: ["get", "list", "watch", "create", "update", "patch"]
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: openebs-zfs-provisioner-binding
subjects:
  - kind: ServiceAccount
    name: openebs-zfs-controller-sa
    namespace: openebs
roleRef:
  kind: ClusterRole
  name: openebs-zfs-provisioner-role
  apiGroup: rbac.authorization.k8s.io
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: openebs-zfs-snapshotter-binding
subjects:
  - kind: ServiceAccount
    name: openebs-zfs-controller-sa
    namespace: openebs
roleRef:
  kind: ClusterRole
  name: openebs-zfs-snapshotter-role
  apiGroup: rbac.authorization.k8s.io
---
# Source: zfs-localpv/templates/rbac.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: openebs-zfs-driver-registrar-binding
subjects:
  - kind: ServiceAccount
    name: openebs-zfs-node-sa
    namespace: openebs
roleRef:
  kind: ClusterRole
  name: openebs-zfs-driver-registrar-role
  apiGroup: rbac.authorization.k8s.io
---
# Source: zfs-localpv/templates/zfs-node.yaml
kind: DaemonSet
apiVersion: apps/v1
metadata:
  name: openebs-zfs-localpv-node
  labels:
    chart: zfs-localpv-1.1.0
    heritage: Helm
    openebs.io/version: "1.1.0"
    role: "openebs-zfs"
    name: "openebs-zfs-node"
    release: openebs
    openebs.io/component-name: "openebs-zfs-node"
spec:
  selector:
    matchLabels:
      name: "openebs-zfs-node"
      release: openebs
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 100%
    type: RollingUpdate
  template:
    metadata:
      labels:
        chart: zfs-localpv-1.1.0
        heritage: Helm
        openebs.io/version: "1.1.0"
        role: "openebs-zfs"
        name: "openebs-zfs-node"
        release: openebs
        openebs.io/component-name: "openebs-zfs-node"
        
        name: openebs-zfs-node
    spec:
      priorityClassName: system-node-critical
      serviceAccount: openebs-zfs-node-sa
      hostNetwork: true
      containers:
        - name: csi-node-driver-registrar
          image: "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0"
          imagePullPolicy: IfNotPresent
          args:
            - "--v=5"
            - "--csi-address=$(ADDRESS)"
            - "--kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)"
          lifecycle:
            preStop:
              exec:
                command: ["/bin/sh", "-c", "rm -rf /registration/zfs-localpv /registration/zfs-localpv-reg.sock"]
          env:
            - name: ADDRESS
              value: /plugin/csi.sock
            - name: DRIVER_REG_SOCK_PATH
              value: /var/lib/kubelet/plugins/zfs-localpv/csi.sock
            - name: KUBE_NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
            - name: NODE_DRIVER
              value: openebs-zfs
          volumeMounts:
            - name: plugin-dir
              mountPath: /plugin
            - name: registration-dir
              mountPath: /registration
        - name: openebs-zfs-plugin
          securityContext:
            privileged: true
            allowPrivilegeEscalation: true
          image: "openebs/zfs-driver:1.1.0"
          imagePullPolicy: IfNotPresent
          args:
            - "--nodeid=$(OPENEBS_NODE_ID)"
            - "--endpoint=$(OPENEBS_CSI_ENDPOINT)"
            - "--plugin=$(OPENEBS_NODE_DRIVER)"
          env:
            - name: OPENEBS_NODE_ID
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
            - name: OPENEBS_CSI_ENDPOINT
              value: unix:///plugin/csi.sock
            - name: OPENEBS_NODE_DRIVER
              value: agent
            - name: OPENEBS_NAMESPACE
              value: openebs
          volumeMounts:
            - name: plugin-dir
              mountPath: /plugin
            - name: device-dir
              mountPath: /dev
            - name: encr-keys
              mountPath: /home/keys
            - name: chroot-zfs
              mountPath: /sbin/zfs
              subPath: zfs
            - name: host-root
              mountPath: /host
              mountPropagation: "HostToContainer"
              readOnly: true
            - name: pods-mount-dir
              mountPath: /var/lib/kubelet/
              # needed so that any mounts setup inside this container are
              # propagated back to the host machine.
              mountPropagation: "Bidirectional"
      volumes:
        - name: device-dir
          hostPath:
            path: /dev
            type: Directory
        - name: encr-keys
          hostPath:
            path: /home/keys
            type: DirectoryOrCreate
        - name: chroot-zfs
          configMap:
            defaultMode: 0555
            name: openebs-zfspv-bin
        - name: host-root
          hostPath:
            path: /
            type: Directory
        - name: registration-dir
          hostPath:
            path: /var/lib/kubelet/plugins_registry/
            type: DirectoryOrCreate
        - name: plugin-dir
          hostPath:
            path: /var/lib/kubelet/plugins/zfs-localpv/
            type: DirectoryOrCreate
        - name: pods-mount-dir
          hostPath:
            path: /var/lib/kubelet/
            type: Directory
---
# Source: zfs-localpv/templates/contoller.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: openebs-zfs-localpv-controller
  labels:
    chart: zfs-localpv-1.1.0
    heritage: Helm
    openebs.io/version: "1.1.0"
    role: "openebs-zfs"
    app: "openebs-zfs-controller"
    release: openebs
    component: "openebs-zfs-controller"
    openebs.io/component-name: "openebs-zfs-controller"
spec:
  selector:
    matchLabels:
      app: "openebs-zfs-controller"
      release: openebs
      component: "openebs-zfs-controller"
  serviceName: "openebs-zfs"
  replicas: 
  template:
    metadata:
      labels:
        chart: zfs-localpv-1.1.0
        heritage: Helm
        openebs.io/version: "1.1.0"
        role: "openebs-zfs"
        app: "openebs-zfs-controller"
        release: openebs
        component: "openebs-zfs-controller"
        openebs.io/component-name: "openebs-zfs-controller"
        
        name: openebs-zfs-controller
    spec:
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: app
                operator: In
                values:
                - openebs-zfs-controller
            topologyKey: "kubernetes.io/hostname"
      priorityClassName: system-cluster-critical
      serviceAccount: openebs-zfs-controller-sa
      containers:
        - name: csi-resizer
          image: "quay.io/k8scsi/csi-resizer:v0.4.0"
          args:
            - "--v=5"
            - "--csi-address=$(ADDRESS)"
            - "--leader-election"
          env:
            - name: ADDRESS
              value: /var/lib/csi/sockets/pluginproxy/csi.sock
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
        - name: csi-snapshotter
          image: "quay.io/k8scsi/csi-snapshotter:v2.0.1"
          imagePullPolicy: IfNotPresent
          args:
            - "--csi-address=$(ADDRESS)"
            - "--leader-election"
          env:
            - name: ADDRESS
              value: /var/lib/csi/sockets/pluginproxy/csi.sock
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
        - name: snapshot-controller
          image: "quay.io/k8scsi/snapshot-controller:v2.0.1"
          args:
            - "--v=5"
            - "--leader-election=true"
          imagePullPolicy: IfNotPresent
        - name: csi-provisioner
          image: "quay.io/k8scsi/csi-provisioner:v1.6.0"
          imagePullPolicy: IfNotPresent
          args:
            - "--csi-address=$(ADDRESS)"
            - "--v=5"
            - "--feature-gates=Topology=true"
            - "--strict-topology"
            - "--enable-leader-election"
            - "--leader-election-type=leases"
            - "--extra-create-metadata=true"
          env:
            - name: ADDRESS
              value: /var/lib/csi/sockets/pluginproxy/csi.sock
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
        - name: openebs-zfs-plugin
          image: "openebs/zfs-driver:1.1.0"
          imagePullPolicy: IfNotPresent
          env:
            - name: OPENEBS_CONTROLLER_DRIVER
              value: controller
            - name: OPENEBS_CSI_ENDPOINT
              value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
            - name: OPENEBS_NAMESPACE
              value: openebs
            - name: OPENEBS_IO_INSTALLER_TYPE
              value: "zfs-operator"
            - name: OPENEBS_IO_ENABLE_ANALYTICS
              value: "true"
          args :
            - "--endpoint=$(OPENEBS_CSI_ENDPOINT)"
            - "--plugin=$(OPENEBS_CONTROLLER_DRIVER)"
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
      volumes:
        - name: socket-dir
          emptyDir: {}
---
# Source: zfs-localpv/templates/csidriver.yaml
# Create the CSI Driver object
apiVersion: storage.k8s.io/v1beta1
kind: CSIDriver
metadata:
  name: zfs.csi.openebs.io
spec:
  # do not require volumeattachment
  attachRequired: false
  podInfoOnMount: false

NOTES:
The OpenEBS ZFS LocalPV has been installed. Check its status by running:
$ kubectl get pods -n openebs -l role=openebs-zfs

For more information, visit our Slack at https://openebs.io/community or view 
the documentation online at http://docs.openebs.io/.


```
**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: deploy/helm/charts/values.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+zfsPlugin:
+  name: "openebs-zfs-plugin"
+  image:
+    # Make sure that registry name end with a '/'.
+    # For example : quay.io/ is a correct value here and quay.io is incorrect
+    registry:
+    repository: openebs/zfs-driver
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: ci
```
Comment (prateekpandey14) [inline deploy/helm/charts/values.yaml:111]: tag needs update: 1.1.0 ?

File: deploy/yamls/zfs-driver.yaml [hunk 2]
```diff
@@ -442,7 +442,7 @@ kind: ServiceAccount
 apiVersion: v1
 metadata:
   name: openebs-zfs-controller-sa
-  namespace: kube-system
```
Comment (pawanpraka1) [inline deploy/yamls/zfs-driver.yaml:445]: let us revert this change and keep using kube-system namespace for now.

**Review by pawanpraka1 — COMMENTED** [review]
@shubham14bajpai not abe to see the k8s snapshots CRs here.
1. volumesnapshots.snapshot.storage.k8s.io
2. volumesnapshotcontents.snapshot.storage.k8s.io
3. volumesnapshotclasses.snapshot.storage.k8s.io

File: deploy/helm/charts/templates/zfs-node.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          hostPath:
+            path: /var/lib/kubelet/plugins_registry/
+            type: DirectoryOrCreate
+        - name: plugin-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins/zfs-localpv/
+            type: DirectoryOrCreate
+        - name: pods-mount-dir
+          hostPath:
+            path: /var/lib/kubelet/
```
Comment (prateekpandey14) [inline deploy/helm/charts/templates/zfs-node.yaml:127]: `/var/lib/kubelet`  we have to make this configurable via values.yaml file

File: deploy/helm/charts/templates/zfs-node.yaml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          configMap:
+            defaultMode: 0555
+            name: openebs-zfspv-bin
+        - name: host-root
+          hostPath:
+            path: /
+            type: Directory
+        - name: registration-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins_registry/
```
Comment (prateekpandey14) [inline deploy/helm/charts/templates/zfs-node.yaml:119]: `/var/lib/kubelet`  we have to make this configurable via values.yaml file

File: deploy/helm/charts/templates/zfs-node.yaml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            type: DirectoryOrCreate
+        - name: plugin-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins/zfs-localpv/
```
Comment (prateekpandey14) [inline deploy/helm/charts/templates/zfs-node.yaml:123]: same as above comment

File: deploy/helm/charts/templates/zfs-node.yaml [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+            - "--kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)"
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sh", "-c", "rm -rf /registration/zfs-localpv /registration/zfs-localpv-reg.sock"]
+          env:
+            - name: ADDRESS
+              value: /plugin/csi.sock
+            - name: DRIVER_REG_SOCK_PATH
+              value: /var/lib/kubelet/plugins/zfs-localpv/csi.sock
```
Comment (prateekpandey14) [inline deploy/helm/charts/templates/zfs-node.yaml:48]: `/var/lib/kubelet` we have to make this configurable via values.yaml file

File: deploy/helm/charts/templates/contoller.yaml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            - name: OPENEBS_CONTROLLER_DRIVER
+              value: controller
+            - name: OPENEBS_CSI_ENDPOINT
+              value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
+            - name: OPENEBS_NAMESPACE
+              value: openebs
+            - name: OPENEBS_IO_INSTALLER_TYPE
+              value: "zfs-operator"
+            - name: OPENEBS_IO_ENABLE_ANALYTICS
+              value: "true"
```
Comment (prateekpandey14) [inline deploy/helm/charts/templates/contoller.yaml:101]: we can make this configurable 
```
            - name: OPENEBS_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: OPENEBS_IO_INSTALLER_TYPE
              value: "zfs-localpv-helm"
            - name: OPENEBS_IO_ENABLE_ANALYTICS
              value: "{{ .Values.analytics.enabled }}"
```

File: deploy/helm/charts/templates/zfs-node.yaml [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+            - name: OPENEBS_NODE_ID
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: OPENEBS_CSI_ENDPOINT
+              value: unix:///plugin/csi.sock
+            - name: OPENEBS_NODE_DRIVER
+              value: agent
+            - name: OPENEBS_NAMESPACE
+              value: openebs
```
Comment (prateekpandey14) [inline deploy/helm/charts/templates/zfs-node.yaml:80]: we can use downward API here

File: deploy/helm/charts/templates/contoller.yaml [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+          imagePullPolicy: {{ .Values.zfsPlugin.image.pullPolicy }}
+          env:
+            - name: OPENEBS_CONTROLLER_DRIVER
+              value: controller
+            - name: OPENEBS_CSI_ENDPOINT
+              value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
+            - name: OPENEBS_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
```
Comment (pawanpraka1) [inline deploy/helm/charts/templates/contoller.yaml:99]: This is the namespace where ZFS_LocalPV CRs will be created. So we can install it in kube-system namespace and by default openebs was used for creating the CRs. Although this will also work.

File: deploy/helm/charts/values.yaml [hunk 10]
```diff
@@ -0,0 +1,135 @@
+# Default values for openebs-zfslocalpv.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+release:
+  version: "1.1.0"
```
Comment (pawanpraka1) [inline deploy/helm/charts/values.yaml:5]: 1.2.1

File: deploy/helm/charts/Chart.yaml [hunk 11]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v2
+name: zfs-localpv
+description: CSI Driver for dynamic provisioning of ZFS Persistent Local Volumes.
+version: 1.2.0
```
Comment (pawanpraka1) [inline deploy/helm/charts/Chart.yaml:4]: 1.2.1?

File: deploy/helm/charts/templates/NOTES.txt [hunk 12]
```diff
@@ -0,0 +1,5 @@
+The OpenEBS ZFS LocalPV has been installed. Check its status by running:
```
Comment (pawanpraka1) [inline deploy/helm/charts/templates/NOTES.txt:1]: once OpenEBS ZFS LocalPV has been installed check its status by running.

Comment (shubham14bajpai) [reply deploy/helm/charts/templates/NOTES.txt:1]: This text is displayed on the terminal at the end when the helm install is completed.

File: deploy/helm/charts/Chart.yaml (hunk 11 above)
Comment (kmova) [reply deploy/helm/charts/Chart.yaml:4]: version is helm chart version. The app version should match the underlying tag versions.

File: deploy/helm/charts/Chart.yaml [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  - cloud-native-storage
+  - block-storage
+  - filesystem
+  - ZFS
+  - Local Persistent Volumes
+  - storage
+sources:
+- https://github.com/openebs/zfs-localpv
+maintainers:
+  - name: kmova
```
Comment (kmova) [inline deploy/helm/charts/Chart.yaml:18]: Probably just keep this to prateek and michael at the moment?

File: deploy/helm/charts/values.yaml [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: 1.2.1
+
+role: openebs-zfs
+
+crd:
+  enableInstall: true
+
+serviceAccount:
+  controller:
```
Comment (kmova) [inline deploy/helm/charts/values.yaml:121]: should this be zfsController to be consistent with zfsNode?

File: deploy/helm/charts/values.yaml [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+  #   cpu: 10m
+  #   memory: 32Mi
+  ## Labels to be added to openebs-zfs controller pods
+  podLabels:
+    name: openebs-zfs-controller
+  nodeSelector: {}
+  tolerations: []
+  securityContext: {}
+
+zfsPlugin:
```
Comment (kmova) [inline deploy/helm/charts/values.yaml:104]: What is this for? csi plugin?

Comment (prateekpandey14) [reply deploy/helm/charts/values.yaml:104]: Yes , its a common tag used in controller and node files

File: deploy/helm/charts/templates/contoller.yaml (hunk 9 above)
Comment (prateekpandey14) [reply deploy/helm/charts/templates/contoller.yaml:99]: Since the zfs localpv is getting deployed in kube-system namespace , then the value here will be `kube-system` not `openebs` , is that fine ?

Also i have added the priorityClass in cstor charts PR , PTAL in case we want to add it here ?

File: deploy/helm/charts/templates/contoller.yaml [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "zfslocalpv.controller.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "zfslocalpv.controller.matchLabels" . | nindent 6 }}
+  serviceName: {{ .Values.serviceAccount.controller.name }}
+  replicas: {{ .Values.controller.replicas }}
```
Comment (pawanpraka1) [inline deploy/helm/charts/templates/contoller.yaml:15]: can we define replicas to 1 in values.yaml.

File: deploy/helm/charts/templates/contoller.yaml [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+  {{- with .Values.controller.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "zfslocalpv.controller.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "zfslocalpv.controller.matchLabels" . | nindent 6 }}
+  serviceName: {{ .Values.serviceAccount.controller.name }}
```
Comment (pawanpraka1) [inline deploy/helm/charts/templates/contoller.yaml:14]: serviceName in the operator yaml is "openebs-zfs", here we are using service account name as service name. I hope this is not an issue cc: @prateekpandey14

File: deploy/helm/charts/templates/contoller.yaml (hunk 9 above)
Comment (pawanpraka1) [reply deploy/helm/charts/templates/contoller.yaml:99]: I would suggest to document this that it will create the resource where we are deployinh the helm chart. In contrast to what zfs operator has where both can be different namespace.

**Review by pawanpraka1 — COMMENTED** [review]
@shubham14bajpai have given some minor comments. Can we install it via helm in travis (we install via operator here https://github.com/openebs/zfs-localpv/blob/master/ci/ci-test.sh#L27) and run the bdd test once for this and see if there are no issues here. Or we can run the bdd locally also.

**Comment by shubham14bajpai** [comment]
> @shubham14bajpai have given some minor comments. Can we install it via helm in travis (we install via operator here https://github.com/openebs/zfs-localpv/blob/master/ci/ci-test.sh#L27) and run the bdd test once for this and see if there are no issues here. Or we can run the bdd locally also.

I was able to run the ci-test successfully on travis as well as locally.

**Review by pawanpraka1 — APPROVED** [review]
looks good. I need one more approval for this PR cc: @prateekpandey14 @kmova

**Comment by akhilerm** [comment]
@shubham14bajpai  Can you make a change in the [release workflow](https://github.com/openebs/zfs-localpv/blob/master/.github/workflows/release.yml#L17), like [this](https://github.com/openebs/node-disk-manager/pull/526/files), so that only v* releases trigger a workflow run and not the chart releases.

File: deploy/helm/charts/README.md [hunk 18]
```diff
@@ -0,0 +1,135 @@
+
+#  OpenEBS LocalPV Provisioner
```
Comment (kmova) [inline deploy/helm/charts/README.md:2]: OpenEBS ZFS Local PV Provisioner

**Review by kmova — APPROVED** [review]
